File: 1323 Maximum 69 Number.java
class Solution {
    public int maximum69Number (int num) {
        return Integer.valueOf(String.valueOf(num).replaceFirst("6", "9"));
    }
}
/* 0 ms solution
class Solution {
    public int maximum69Number (int num) {
        int i = 1, n = num, i6 = -1;
        while (n > 0) {
            if (n % 10 == 6) i6 = i;
            n /= 10;
            i *= 10;
        }

        return (i6 == -1)? num : num + 3 * i6;        
    }
}*/

File: 1323 Maximum 69 Number.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        return int(str(num).replace('6', '9', 1))
'''
class Solution:
    def maximum69Number (self, num: int) -> int:
        max_val = num
        length = len(str(num))
        for i in range(length):
            temp = list(str(num))
            temp[i] = '9'
            max_val = max(max_val, int(''.join(temp)))
        return max_val

class Solution:
    def maximum69Number (self, num: int) -> int:
        i, n, i6 = 1, num, -1
        while n > 0:
            if n % 10 == 6:
                i6 = i
            n //= 10
            i *= 10
        
        return num if i6 == -1 else num + 3 * i6
'''